### DISCUSSION TITLE
Function that Defines Diffusion Interior to a Domain

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#24574] [Q&A Modules: General] [answered]
Is there an example problem where in 2D a function is defined for a circle or any shape interior to a rectangular domains or similar,where the diffusion is defined by a function of the equation of a circle or any other shape, to be a constant value (by function not mesh shape) interior to the rectangular domain with source and sink boundary conditions defined to drive diffusion through the rectangular domain?

**Comment by GiudGiud** [comment, marked as answer]
Hello

This seems pretty specific so I m going to say no.
If you want a circle mesh with a rectangle around you can use that:
https://mooseframework.inl.gov/source/meshgenerators/ConcentricCircleMeshGenerator.html
then stitch a few rectangular meshes around it with those
https://mooseframework.inl.gov/source/meshgenerators/CartesianMeshGenerator.html
https://mooseframework.inl.gov/source/meshgenerators/StitchedMeshGenerator.html

To define a function only in a circle, you can use a ParsedFunction with a simple `if(x*x + y*y < r*r, this_function, 0)`

With the mesh generators I pasted you can define different subdomains so you can block restrict source and sink

Guillaume
